src/game/lara/lara_hair.c

```diff
@@ -104,7 +112,8 @@ void Lara_Hair_Control(bool in_cutscene)
 
     // hips
     Matrix_Push();
-    objptr = g_Lara.mesh_ptrs[LM_HIPS];
+    objptr = in_cutscene ? g_Meshes[object->mesh_index + LM_HIPS]
+                         : g_Lara.mesh_ptrs[LM_HIPS];
```

Comment:
Would it be possible to always use `object` and never refer to `g_Lara` directly, or at most once? I'd like to keep the `if (in_cutscene)` check to minimum or ideally do without it altogether.

Reply:
> Would it be possible to always use `object` and never refer to `g_Lara` directly, or at most once? I'd like to keep the `if (in_cutscene)` check to minimum or ideally do without it altogether.

Yes, no problem to make a `mesh_base` of sorts, then it's a single check at the beginning for the Lara type. We do still need `g_Lara.mesh_ptrs` to track normal in-game mesh swaps, so the check would set `mesh_base` to either these pointers or the relevant position in `g_Meshes`.

We could also change `in_cutscene` to a local variable in the function - or even a module variable - and remove it as an argument - so `in_cutscene = m_LaraType != O_LARA` ? It's still needed further down for the water level checks, but at least it would mean all calls to `Lara_Hair_Control` would be the same.

Reply:
>We could also change in_cutscene to a local variable in the function - or even a module variable - and remove it as an argument - so in_cutscene = m_LaraType != O_LARA ? It's still needed further down for the water level checks, but at least it would mean all calls to Lara_Hair_Control would be the same.

Yes I'd prefer it to be this way.